### ISSUE TITLE
Could not set up HTTP server on lhost / can't transfer payload

### ISSUE THREAD

**Issue by jansramek** [threat9/routersploit#550]
## Current Behavior
* Crashes when attempting to transfer payload/setting HTTP server
* Same behaviour on Mac OS & Kali Linux

`cmd (MIPSLE Reverse TCP) > run
[*] Using wget method
[-] Could not set up HTTP Server on 192.168.0.102:5555
[-] Exploit failed to transfer payload`

## Expected Behavior
* Should open reverse shell

## Steps to Reproduce (for bugs)
1. use exploits/routers/linksys/eseries_themoon_rce
2. set target 192.168.0.1
3. run
4. set payload reverse_tcp
5. set lhost 192.168.0.102
6. run

## Environment
* RouterSploit Version used:
* Operating System and version: Mac OS 10.14.1 | Kali Linux 2018.1
* Python Version: 3.7.1 | 3.6.7
* Python Environment: `asn1crypto==0.24.0
bcrypt==3.1.4
certifi==2018.10.15
cffi==1.11.5
chardet==3.0.4
cryptography==2.3.1
future==0.16.0
idna==2.7
paramiko==2.4.2
ply==3.11
pyasn1==0.4.4
pycparser==2.19
pycryptodome==3.6.6
pycryptodomex==3.6.6
PyNaCl==1.3.0
pysmi==0.3.1
pysnmp==4.4.6
requests==2.20.0
six==1.11.0
urllib3==1.24`

**Comment by stefan9999991** [comment]
have the exact same problem! Please notify me if you find something that helps you.

**Comment by TheProgrammingM** [comment]
Got the same problem, please notify me tooo (:

**Comment by stefan9999991** [comment]
Any solutions yet?

**Comment by GianisTsol** [comment]
Also having it :( notif me when you find a solution

**Comment by Keskebeu** [comment]
Same here.

**Comment by Bbcomb** [comment]
same lol.

**Comment by DougP19** [comment]
Exact same here. However I can verify the router I'm scanning is NOT a Linksys router.

**Comment by jansramek** [comment]
> I believe this has to do with the type of Linksys router. It's probably a different type of Linksys router.

Same on non Linksys router. It is probably false positives.

**Comment by Keskebeu** [comment]
Could be the case. The router I tried it on was a linksys router. When I
used it in a Motorola router nothing came up. I also read somewhere else
that it had to with that and we are expected to change the code based on
the linksys router.

It could be a router which is made by Linksys or uses Linksys components
and or software, just different name.

**Comment by fahd93** [comment]
same here

**Comment by JakeFrom98** [comment]
Me too...

**Comment by JakeFrom98** [comment]
I'm also having that issue but I think I might have found a fix keyword "might" make sure apache is started by typing "sudo service apache2 start"

**Comment by Hayzr** [comment]
Looks like a false positive. If you go through the logs it uses an HTTP GET for the tmUnblock.cgi file. It marks a successful GET as Vulnerable. 

You can wget the file yourself: wget <router IP>/tmUnblock.cgi
Compare it to the source code for the login screen. In my case, they're the same.

**Comment by HarizDharma** [comment]
same here. i got this problem, how to solve ?

**Comment by ErAle001** [comment]
I also have the same problem. Although it marks the router as vulnerable, I can confirm that the router is not Linksys, so it might me a false positive but I wonder why the HTTP server error occurs. 
Setting up the server on lhost shouldnt be related to the payload working or not against the router.
Any thoughts on this?

The fact of this issue being open since 2018 is not a good sign.
![Captura de pantalla de 2020-04-16 15-44-01](https://user-images.githubusercontent.com/59049954/79463458-2acec680-7ff9-11ea-99cf-5cb110758333.png)

**Comment by LogicatedDev** [comment]
i get this too

**Comment by sealoomaan** [comment]
Anybody here found the solution ?

**Comment by r4ldin** [comment]
Same error, but if I set a port that the router is using for other tcp/udp connections the payload is being handled somehow, i think anyway, 
The router is a Huawei DG8245W2-10, not mine so I don't know fw v, the owner have forgotten the admin passw.

Here is the handshake - payload
scr->wireshark
![mp](https://user-images.githubusercontent.com/74553087/100383911-1d61cc00-301f-11eb-802a-79777f2c6826.png)

At first glance the response looked like a normal POST to /tmUnblock.cgi  
....but there is something
![mpr](https://user-images.githubusercontent.com/74553087/100384233-f6f06080-301f-11eb-94c3-ac139e859c0e.png)

![payload](https://user-images.githubusercontent.com/74553087/100384391-59496100-3020-11eb-9f96-d8dc4983d40a.png)

![mp2](https://user-images.githubusercontent.com/74553087/100384421-6b2b0400-3020-11eb-8995-cdf8d500da7c.png)

**Comment by gabriel2018-27** [comment]
![image](https://user-images.githubusercontent.com/75048317/100484776-79d6f100-30dc-11eb-8da2-35acc7d2724f.png)

heeeeeeeelp!!

**Comment by blackerr** [comment]
any solution? 
![image](https://user-images.githubusercontent.com/36746171/100530014-40da7180-31ed-11eb-88a2-54a4544a7de3.png)

**Comment by OmiceyO** [comment]
2021 still doesnt work. Probably as the guy send above its a "False Positive"

**Comment by cm038** [comment]
> any solution?
> ![image](https://user-images.githubusercontent.com/36746171/100530014-40da7180-31ed-11eb-88a2-54a4544a7de3.png)

I have same problem

**Comment by cm038** [comment]
Nobody ?

**Comment by cm038** [comment]
Hey... When i chance the lport it comes step closer 

cmd (MIPSLE Bind TCP) > set lport 8080                    
lport => 8080                                                
cmd (MIPSLE Bind TCP) > run                              

[*] Using wget method                                        
[*] Using wget to download binary                            
[-] Exploit failed to transfer payload

**Comment by 0vertime-dev** [comment]
yeah still getting this, no idea whats going on, false-positive seems the most likely

**Comment by imansour12** [comment]
getting this too

**Comment by andvargrad** [comment]
![Screenshot_462](https://user-images.githubusercontent.com/62504766/132132565-2a53fc1b-3364-453d-9dfa-690207efa50f.jpg)


I have same problem

**Comment by Codeiology** [comment]
I tried the checker for this exploitation module on a non-link sys router, and it showed up as vulnerable, but had the same issue. I think we can confirm this is a false positive now.

**Comment by Dlazder** [comment]
![Screenshot_2023-09-10-11-51-04-304_com.termux.jpg](https://github.com/threat9/routersploit/assets/124579368/e889e68b-4e6b-4c98-b0b8-b7fa9f2238d6)

I have the same problem(

**Comment by Munkhbadral1** [comment]
same

**Comment by cankaygisiz** [comment]
![image_2024-01-01_230329770](https://github.com/threat9/routersploit/assets/52928009/09775ec0-2336-4231-b703-d788a2713110)
any update for this problem?

**Comment by iqnnis2hacker** [comment]
had this  also

**Comment by Mohamed3bdelwahab** [comment]
still issue exist  is there any working methods
